src/core/client.ts

```diff
@@ -0,0 +1,180 @@
+import { Cryptography as crypto } from './cryptography';
+import * as rp from 'request-promise';
+import * as _ from 'underscore';
+import * as qs from 'querystring';
+import * as elliptic from 'elliptic';
+import { Invoice } from '../models/invoice';
+import { Rate } from '../models/rate';
+
+export interface IBTCPayClient {
+  get_rates(currencyPairs, storeID): Promise<Rate[]>;
+  create_invoice(payload: any, token?: any): Promise<Invoice>;
+  get_invoice(invoice_id: string, token?: any): Promise<Invoice[]>;
+  get_invoices(params: any, token?: any): Promise<Invoice[]>;
+}
+
+export class BTCPayClient implements IBTCPayClient {
```

Comment:
I am not fan of creating an interface which has no concrete use.

Reply:
The thought behind that was to use it for tests & mocking. But its ok for me if we kill it. I am using inversify as dependecy injector in one of my projects and there I am quite often use this kind of strategy.

Reply:
> Awesome!!!! there is a conflict though.

I don't have write access to see the conflict and resolve it.

Reply:
but client.js should not be existent anymore...

Reply:
Yeah, I prefer you remove the interface.
1. This is a pain to mock as everytimes a new method is added, the mock need to be udpated
2. This is a pain to maintain as we need to update interface and concrete class
3. This is not really needed, afaik typescript support some sort of duck typing so if your project really need it, you can do it easily and only for the methods you really care.

Reply:
done removed it.